src/daft-core/src/lit/python.rs

```diff
@@ -649,10 +649,22 @@ fn pydecimal_to_decimal_lit(ob: &Bound<PyAny>) -> PyResult<Literal> {
 
 fn numpy_array_like_to_tensor_lit(ob: &Bound<PyAny>) -> PyResult<Literal> {
     let py = ob.py();
-    let np_asarray = py
-        .import(intern!(py, "numpy"))?
-        .getattr(intern!(py, "asarray"))?;
-    let ob = np_asarray.call1((ob,))?;
+    let ob = match py.import(intern!(py, "cupy")) {
+        Ok(cupy_module) => {
+            let cupy_ndarray_ty = cupy_module.getattr(intern!(py, "ndarray"))?;
+            if ob.is_instance(&cupy_ndarray_ty)? {
+                cupy_module.getattr(intern!(py, "asnumpy"))?.call1((ob,))?
+            } else {
+                py.import(intern!(py, "numpy"))?
+                    .getattr(intern!(py, "asarray"))?
+                    .call1((ob,))?
+            }
+        }
+        Err(_) => py
+            .import(intern!(py, "numpy"))?
+            .getattr(intern!(py, "asarray"))?
+            .call1((ob,))?,
```

Comment:
Instead of having special logic in this function to handle cupy arrays, let's create a new function called something like `cupy_array_to_tensor_lit` and call that when we get a cupy array instance.